### DISCUSSION TITLE
Text filtering in more windows

### DISCUSSION THREAD

**Discussion by perezdidac** [OpenTTD/OpenTTD#8723] [Features]
Continuing the work done in https://github.com/OpenTTD/OpenTTD/pull/8603 and https://github.com/OpenTTD/OpenTTD/pull/8706, I want to keep adding filters where players find pain going through a list.

I want to start this discussion thread to understand what are players looking for. Some ideas:

- New Rail Vehicle window: filtering by train name
- Active NewGRF files in the NewGRF Settings window.

**Comment by 2TallTyler** [comment]
I think the vehicle buy menu would be most useful.

I was surprised that the industry directory doesn't already have a text filter, like the town window.

The company station list can also get rather lengthy, although I personally never use it to find a station.

**Comment by James103** [comment]
The vehicle lists (4 buttons on the top bar just before the zoom in/out buttons) could use a text filter, to filter vehicles by their name. A second text filter could be added above the list of groups in order to filter groups by their name.

**Comment by LC-Zorg** [comment]
>New Rail Vehicle window: filtering by train name

I would probably never use it. Even playing with 2CC Trains. I don't rule out that in some situations it would be useful for some players, but it seems quite niche to me. I have a reservation here mainly because there is no good place for the filter in this layout. Some cargo names can be long, so the filter would have to form another bar. It would add an option that I think would be very rarely used, and it would increase the size of the window unnecessarily.
However, I think there could be a place for this filter. Like the finance window, you could make a button that changes the window layout by adding a more extensive buy menu with more filters, including a text filter.

>Active NewGRF files in the NewGRF Settings window.

I would prefer the categorization options here - it would make it much easier to find each add-on and spot others that you didn't know about. Of course, it wouldn't be that simple, as it would require you to assign a category to each add-on, but that would give you much better options, especially on the left side of the window and in the addon search window.

>A second text filter could be added above the list of groups in order to filter groups by their name.

I sincerely doubt the usefulness of this feature here. I have quite a few online game saves, and I've never really seen anyone have more than a dozen groups. This amount is visible without scrolling through the list, so why a filter?

>I was surprised that the industry directory doesn't already have a text filter, like the town window.

It wouldn't be bad. Moving the cargo filter to the beam below would reduce the width of the window. Then there would be a place in the upper right corner for the text filter. It is a good proposition for me. :)

Perhaps a good solution would be to condition the appearance of the filter on the amount of the list length? For example, why a filter when only 10 vehicles or 2 groups of stations are available?
